scripts/locations_data_load/test/test_locations_lambda.py

```diff
@@ -0,0 +1,176 @@
+    def test_update_records(self, mock_dynamodb_resource):
+        # Mock DynamoDB resource and tables
+        mock_dynamodb = mock_dynamodb_resource.return_value
+        mock_org_table = Mock()
+        mock_locations_table = Mock()
+        mock_dynamodb.Table.side_effect = (
+            lambda table_name: mock_org_table
+            if table_name == "organisations"
+            else mock_locations_table
+        )
+
+        # Mock DynamoDB scan responses
+        mock_org_response = {
+            "Items": [{"id": "org_id_1", "identifier": {"value": "123"}}]
+        }
+        mock_locations_response = {
+            "Items": [
+                {"id": "loc_id_1", "managingOrganization": "", "lookup_field": "123"}
+            ]
+        }
+        mock_org_table.scan.return_value = mock_org_response
+        mock_locations_table.scan.return_value = mock_locations_response
+
+        # Call the function
+        update_records()
+
+        # Assert that update_item was called with the correct arguments
+        mock_locations_table.update_item.assert_called_once_with(
+            Key={"id": "loc_id_1"},
+            UpdateExpression="SET managingOrganization = :val",
+            ExpressionAttributeValues={":val": "org_id_1"},
+        )
+
+
+class TestReadODSAPI(unittest.TestCase):
+    @patch("locations_lambda.requests.get")
+    def test_read_ods_api_success(self, mock_requests_get):
+        params = {"param1": "value1", "param2": "value2"}
+        # Mocking get_api_token to return a dummy token
+        headers = {"Authorization": "Bearer dummy_token"}
+        # Mocking requests.get to return a successful response
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {"data": "dummy_data"}
+        mock_requests_get.return_value = mock_response
+        # Call the function with dummy parameters
+        result = read_ods_api("dummy_api_endpoint", headers, params)
+
+        # Assert that requests.get was called with the expected arguments
+        mock_requests_get.assert_called_once_with(
+            "dummy_api_endpoint",
+            headers={"Authorization": "Bearer dummy_token"},
+            params={"param1": "value1", "param2": "value2"},
+        )
+        # Assert that the result is as expected
+        self.assertEqual(result, {"data": "dummy_data"})
+
+    @patch("locations_lambda.requests.get")
+    def test_read_ods_api_failure(self, mock_requests_get):
+        params = {"param1": "value1", "param2": "value2"}
+        headers = {"Authorization": "Bearer dummy_token"}
+
+        # Mocking requests.get to return a failed response
+        mock_response = MagicMock()
+        mock_response.status_code = 404
+        mock_response.text = "Not Found"
+        mock_requests_get.return_value = mock_response
+
+        # Call the function with dummy parameters
+        result = read_ods_api("dummy_api_endpoint", headers, params)
```

Comment:
Nice - code coverage shows we haven't covered the exception handling code in read-ods_api ... could we maybe add another one to test that ?